### PR TITLE
fix: better handle K8s client error while watching

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
fix: better handle K8s client error while watching.
Screenshot_20210908_140645

### What issues does this PR fix or reference?
N/A

### CODE REVIEW

**Review by olexii4 — APPROVED** [review]
LGTM

File: packages/dashboard-backend/src/devworkspace-client/services/api/workspace-api.ts [hunk 1]
```diff
@@ -188,10 +187,19 @@ export class DevWorkspaceApi implements IDevWorkspaceApi {
       } else {
         callbacks.onError(`Error: Unknown type '${type}'.`);
       }
-    }, error => {
-      console.error('Watch in namespace error:', error);
-      const message = typeof (error) === 'string' ? error : 'Unknown type error.'
-      callbacks.onError(message);
+    }, (error: any) => {
+      let message;
```
Comment (olexii4) [inline packages/dashboard-backend/src/devworkspace-client/services/api/workspace-api.ts:191]: Could we use our ```getMessage(error: unknown): string``` here?
https://github.com/eclipse-che/che-dashboard/blob/main-7.37/packages/common/src/helpers/errors.ts#L21

Comment (sleshchenko) [reply packages/dashboard-backend/src/devworkspace-client/services/api/workspace-api.ts:191]: good idea

Comment (sleshchenko) [reply packages/dashboard-backend/src/devworkspace-client/services/api/workspace-api.ts:191]: The only problem I noticied with it:
https://github.com/eclipse-che/che-dashboard/blob/main-7.37/packages/common/src/helpers/errors.ts#L62
```
 return 'Unexpected error. Check DevTools console and network tabs for more information.'
```
which is good error for frontend but not for backend, which is my case.

leaving as is and will think how we can handle it

**Pull request merged**